src/boot/bootctl-status.c

```diff
@@ -503,6 +504,243 @@ int verb_status(int argc, char *argv[], void *userdata) {
         return r;
 }
 
+static int ref_file(Hashmap *known_files, const char *fn, int increment) {
+        char *k = NULL;
+        int n, r;
+
+        assert(known_files);
+
+        /* just gracefully ignore this. This way the caller doesn't
+           have to verify whether the bootloader entry is relevant */
+        if (!fn)
+                return 0;
+
+        n = PTR_TO_INT(hashmap_get2(known_files, fn, (void**)&k));
+        n += increment;
+
+        assert(n >= 0);
+
+        if (!n) {
```

Comment:
we generally use C's downgrade-to-boolean feature only for stuff that is really boolean-like (and for pointers), but not for arithmetic values. This one is an integer counter, hence please do an explicit zero comparison, i.e. literally spell out `n == 0` rather than `!n`